sqlglot/generator.py

```diff
@@ -2074,6 +2074,11 @@ def intdiv_sql(self, expression: exp.IntDiv) -> str:
     def dpipe_sql(self, expression: exp.DPipe) -> str:
         return self.binary(expression, "||")
 
+    def safedpipe_sql(self, expression: exp.SafeDPipe) -> str:
+        if self.STRICT_STRING_CONCAT:
+            return self.func("CONCAT", *[exp.cast(e, "text") for e in expression.flatten()])
```

Comment:
change to gen (

Reply:
is there a difference in the semantics? I thought it'd make no difference since we expand it to match the `*args`

Reply:
one creates a list, the other is an iterator. so you don’t incur the overhead of creating a list

Reply:
I see, makes sense, thanks for clarifying

Reply:
Done. FYI I also fixed https://github.com/tobymao/sqlglot/pull/1738/commits/14f2386ca447e5409c31d1d3c8e9ed001aff007b#diff-362498362ae539e216cd83ebfe510977107648742f1a1bee7b2db7b5b7d27974R1747 similarly.